### PR TITLE
Moar unit tests

### PR DESCRIPTION
**Pull request by realFlowControl**
This pull request adds more unit tests to bump the code coverage

### CODE REVIEW

File: tests/unit/Framework/MockObject/GeneratorTest.php [hunk 1]
```diff
@@ -45,6 +45,41 @@ public function testGetMockFailsWhenInvalidFunctionNameIsPassedInAsAFunctionToMo
         $this->generator->getMock(stdClass::class, [0]);
     }
 
+    public function testGetMockFailsWithInvalidClass(): void
```
Comment (oliverklee) [inline tests/unit/Framework/MockObject/GeneratorTest.php]: Can we be more specific in the test method names what "invalid" means (e.g., final)?

File: tests/unit/Framework/MockObject/GeneratorTest.php [hunk 2]
```diff
@@ -45,6 +45,41 @@ public function testGetMockFailsWhenInvalidFunctionNameIsPassedInAsAFunctionToMo
...
+    }
+
+    public function testGetMockFailsWithNonExistingClasses(): void
+    {
+        $this->expectException(\PHPUnit\Framework\MockObject\RuntimeException::class);
+
+        $this->generator->getMock(['Tux', 'Bar'], [], [], '', true, true, false, true, false, null, false);
+    }
+
+    public function testGetMockWithExistingClassAsMockName(): void
```
Comment (oliverklee) [inline tests/unit/Framework/MockObject/GeneratorTest.php]: I propose to add `…throwsExcepction` to the corresponding test method names so we have a kind of sentence describing the behavior.

File: tests/unit/Framework/MockObject/GeneratorTest.php [hunk 3]
```diff
@@ -240,4 +296,18 @@ public function testVariadicArgumentsArePassedToMockedMethod(): void
 
         $mock->foo(...$arguments);
     }
+
+    public function testGetClassMethodsWithNonExistingClass(): void
+    {
+        $this->expectException(\PHPUnit\Framework\MockObject\RuntimeException::class);
+
+        $this->generator->getClassMethods('Tux');
```
Comment (oliverklee) [inline tests/unit/Framework/MockObject/GeneratorTest.php]: Should we assert first (in the test) that the class really does not exist?

Comment (flow-control) [reply tests/unit/Framework/MockObject/GeneratorTest.php]: good idea, fixed that

File: tests/unit/Framework/MockObject/GeneratorTest.php (hunk 2 above)
Comment (flow-control) [reply tests/unit/Framework/MockObject/GeneratorTest.php]: done

File: tests/unit/Framework/MockObject/GeneratorTest.php (hunk 1 above)
Comment (flow-control) [reply tests/unit/Framework/MockObject/GeneratorTest.php]: done

**Pull request merged**